File: 0011xx/001131/java/Solution.java
import java.lang.AssertionError;
import java.util.*;
import java.util.regex.*;

class Solution {
    public int maxAbsValExpr(int[] arr1, int[] arr2) {
        int ans = 0;
        for(int j=0;j<arr2.length;++j){
            for(int i=0;i<j;++i){
                int v = 0;
                v += Math.abs(arr1[i]-arr1[j]);
                v += Math.abs(arr2[i]-arr2[j]);
                v += Math.abs(i-j);
                if(v<ans)continue;
                ans=v;
            }
        }
        return ans;
    }
}

File: 0011xx/001131/kotlin/Solution.kt
class Solution {
    fun maxAbsValExpr(arr1: IntArray, arr2: IntArray): Int {
        val size=arr1.size
        var ret=0
        
        var l0 =  arr1[0]+arr2[0]
        var l1 =  arr1[0]-arr2[0]
        var l2 = -arr1[0]+arr2[0]
        var l3 = -arr1[0]-arr2[0]

        for(i in 1..(size-1)){
          ret = max(ret,l0-arr1[i]-arr2[i]+i)
          ret = max(ret,l1-arr1[i]+arr2[i]+i)
          ret = max(ret,l2+arr1[i]-arr2[i]+i)
          ret = max(ret,l3+arr1[i]+arr2[i]+i)
          l0 = max(l0, arr1[i]+arr2[i]-i)
          l1 = max(l1, arr1[i]-arr2[i]-i)
          l2 = max(l2,-arr1[i]+arr2[i]-i)
          l3 = max(l3,-arr1[i]-arr2[i]-i)
        }
        return ret
    }
    
    fun max(a:Int,b:Int):Int{
        return kotlin.math.max(a,b)
    }
}
